Newer JNA versions (dependency from Jline) prevents native builds from working

We cannot upgrade past JNA version 5.11.0 because it prevents the generation of native builds. Trying to generate builds with newer JNA versions, such as 5.17.0, cause the build to fail with: 

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /Users/opiske/.sdkman/candidates/java/21.0.2-graalce/bin/native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dlogging.initial-configurator.min-level=500 -J-Duser.language=en -J-Duser.country=US -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -H:+UnlockExperimentalVMOptions -H:IncludeLocales=en-US -H:-UnlockExperimentalVMOptions -J-Dfile.encoding=UTF-8 -J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk=ALL-UNNAMED --features=io.quarkus.runner.Feature,io.quarkus.runtime.graal.DisableLoggingFeature,io.quarkus.runtime.graal.SkipConsoleServiceProvidersFeature -J--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED -J--add-exports=java.security.jgss/sun.security.jgss=ALL-UNNAMED -J--add-opens=java.base/java.text=ALL-UNNAMED -J--add-opens=java.base/java.io=ALL-UNNAMED -J--add-opens=java.base/java.lang.invoke=ALL-UNNAMED -J--add-opens=java.base/java.util=ALL-UNNAMED -H:+UnlockExperimentalVMOptions -H:BuildOutputJSONFile=cli-0.0.6-SNAPSHOT-runner-build-output-stats.json -H:-UnlockExperimentalVMOptions -H:+UnlockExperimentalVMOptions -H:+GenerateBuildArtifactsFile -H:-UnlockExperimentalVMOptions --strict-image-heap --initialize-at-run-time=org.jline.nativ.Kernel32 --initialize-at-run-time=org.jline.nativ.Kernel32\$CONSOLE_SCREEN_BUFFER_INFO --initialize-at-run-time=org.jline.nativ.Kernel32\$CHAR_INFO --initialize-at-run-time=org.jline.nativ.Kernel32\$MOUSE_EVENT_RECORD --initialize-at-run-time=org.jline.nativ.Kernel32\$FOCUS_EVENT_RECORD --initialize-at-run-time=org.jline.nativ.Kernel32\$COORD --initialize-at-run-time=org.jline.nativ.Kernel32\$KEY_EVENT_RECORD --initialize-at-run-time=org.jline.nativ.Kernel32\$SMALL_RECT --initialize-at-run-time=org.jline.nativ.Kernel32\$WINDOW_BUFFER_SIZE_RECORD --initialize-at-run-time=org.jline.nativ.Kernel32\$INPUT_RECORD --initialize-at-run-time=org.jline.nativ.Kernel32\$MENU_EVENT_RECORD --initialize-at-run-time=org.jline.nativ.JLineLibrary -H:ReflectionConfigurationFiles=reflection-config.json -H:IncludeResourceBundles=consoleui_messages -H:ResourceConfigurationFiles=resources-config.json -H:+UnlockExperimentalVMOptions -H:+AllowFoldMethods -H:-UnlockExperimentalVMOptions -J-Djava.awt.headless=true --no-fallback --link-at-build-time -H:+UnlockExperimentalVMOptions -H:+ReportExceptionStackTraces -H:-UnlockExperimentalVMOptions -H:-AddAllCharsets --enable-url-protocols=http,https --enable-monitoring=heapdump -H:+UnlockExperimentalVMOptions -H:-UseServiceLoaderFeature -H:-UnlockExperimentalVMOptions -J--add-exports=org.graalvm.nativeimage/org.graalvm.nativeimage.impl=ALL-UNNAMED --exclude-config io\.netty\.netty-codec /META-INF/native-image/io\.netty/netty-codec/generated/handlers/reflect-config\.json --exclude-config io\.netty\.netty-handler /META-INF/native-image/io\.netty/netty-handler/generated/handlers/reflect-config\.json cli-0.0.6-SNAPSHOT-runner -jar cli-0.0.6-SNAPSHOT-runner.jar
Warning: The option '-H:IncludeResourceBundles=consoleui_messages' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:ReflectionConfigurationFiles=reflection-config.json' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:ResourceConfigurationFiles=resources-config.json' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it. The build output lists all active experimental options, including where they come from and possible alternatives. If you think an experimental option should be considered as stable, please file an issue.
========================================================================================================================
GraalVM Native Image: Generating 'cli-0.0.6-SNAPSHOT-runner' (executable)...
========================================================================================================================
[1/8] Initializing...                                                                                    (5.1s @ 0.21GB)
 Java version: 21.0.2+13, vendor version: GraalVM CE 21.0.2+13.1
 Graal compiler: optimization level: 2, target machine: armv8-a
 C compiler: cc (apple, arm64, 17.0.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 4 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - io.quarkus.runner.Feature: Auto-generated class by Quarkus from the existing extensions
 - io.quarkus.runtime.graal.DisableLoggingFeature: Adapts logging during the analysis phase
 - io.quarkus.runtime.graal.SkipConsoleServiceProvidersFeature: Skip unsupported console service providers when quarkus.native.auto-service-loader-registration is false
------------------------------------------------------------------------------------------------------------------------
 11 experimental option(s) unlocked:
 - '-H:+AllowFoldMethods' (origin(s): command line)
 - '-H:JNIConfigurationResources' (origin(s): 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-native-3.30.4.jar')
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:-UseServiceLoaderFeature' (origin(s): command line)
 - '-H:ReflectionConfigurationFiles' (origin(s): command line)
 - '-H:ResourceConfigurationFiles' (origin(s): command line)
 - '-H:IncludeResourceBundles' (origin(s): command line)
 - '-H:ResourceConfigurationResources' (origin(s): 'META-INF/native-image/org.jline/jline-terminal-jni/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-jna/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-jansi/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-ffm/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jansi-core/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-terminal-3.30.4.jar', 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-native-3.30.4.jar')
 - '-H:+GenerateBuildArtifactsFile' (origin(s): command line)
 - '-H:AddOpens' (alternative API option(s): --add-opens java.base/java.lang=ALL-UNNAMED; origin(s): command line)
 - '-H:ReflectionConfigurationResources' (origin(s): 'META-INF/native-image/org.jline/jline-terminal-jni/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-jna/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-jansi/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal-ffm/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-3.30.4.jar', 'META-INF/native-image/org.jline/jline-terminal/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-terminal-3.30.4.jar', 'META-INF/native-image/org.jline/jline-native/native-image.properties' in 'file:///Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/lib/org.jline.jline-native-3.30.4.jar')
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 24.18GB of memory (75.6% of 32.00GB system memory, determined at start)
 - 12 thread(s) (100.0% of 12 available processor(s), determined at start)
[2/8] Performing analysis...  [*****]                                                                   (14.3s @ 1.21GB)
   14,108 reachable types   (88.1% of   16,015 total)
   20,837 reachable fields  (59.5% of   35,005 total)
   74,107 reachable methods (59.9% of  123,795 total)
    4,380 types,   624 fields, and 5,856 methods registered for reflection
       91 types,   300 fields, and   138 methods registered for JNI access
        5 native libraries: -framework CoreServices, -framework Foundation, dl, pthread, z

Error: Unsupported features in 2 methods
Detailed message:
Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field com.sun.jna.internal.Cleaner.referenceQueue of constant
    com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3
  scanning root com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3 embedded in
    com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44)
  parsing method com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44) reachable via the parsing context
    at static root method.(Unknown Source)

Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field java.lang.ref.Reference.queue of constant
    com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8: com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8
  reading field com.sun.jna.NativeLibrary.cleanable of constant
    com.sun.jna.NativeLibrary@193ee02e: Native Library <libc.dylib@13159884584>
  reading field com.sun.jna.Library$Handler.nativeLibrary of constant
    com.sun.jna.Library$Handler@144a3407: com.sun.jna.Library$Handler@144a3407
  reading field java.lang.reflect.Proxy.h of constant
    jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584>
  scanning root jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584> embedded in
    org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:51)
  parsing method org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:48) reachable via the parsing context
    at org.jline.terminal.impl.jna.JnaNativePty.open(JnaNativePty.java:63)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.open(JnaTerminalProvider.java:44)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.newTerminal(JnaTerminalProvider.java:199)
    at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:814)
    at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:672)
    at ai.wanaku.cli.main.support.WanakuPrinter.terminalInstance(WanakuPrinter.java:136)
    at ai.wanaku.cli.main.support.ResponseHelper.commonResponseErrorHandler(ResponseHelper.java:12)
    at ai.wanaku.cli.main.commands.resources.ResourcesRemove.doCall(ResourcesRemove.java:38)
    at root method.(Unknown Source)


com.oracle.svm.core.util.UserError$UserException: Unsupported features in 2 methods
Detailed message:
Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field com.sun.jna.internal.Cleaner.referenceQueue of constant
    com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3
  scanning root com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3 embedded in
    com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44)
  parsing method com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44) reachable via the parsing context
    at static root method.(Unknown Source)

Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field java.lang.ref.Reference.queue of constant
    com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8: com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8
  reading field com.sun.jna.NativeLibrary.cleanable of constant
    com.sun.jna.NativeLibrary@193ee02e: Native Library <libc.dylib@13159884584>
  reading field com.sun.jna.Library$Handler.nativeLibrary of constant
    com.sun.jna.Library$Handler@144a3407: com.sun.jna.Library$Handler@144a3407
  reading field java.lang.reflect.Proxy.h of constant
    jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584>
  scanning root jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584> embedded in
    org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:51)
  parsing method org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:48) reachable via the parsing context
    at org.jline.terminal.impl.jna.JnaNativePty.open(JnaNativePty.java:63)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.open(JnaTerminalProvider.java:44)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.newTerminal(JnaTerminalProvider.java:199)
    at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:814)
    at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:672)
    at ai.wanaku.cli.main.support.WanakuPrinter.terminalInstance(WanakuPrinter.java:136)
    at ai.wanaku.cli.main.support.ResponseHelper.commonResponseErrorHandler(ResponseHelper.java:12)
    at ai.wanaku.cli.main.commands.resources.ResourcesRemove.doCall(ResourcesRemove.java:38)
    at root method.(Unknown Source)


	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:85)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:248)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:814)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:592)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:550)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:539)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:721)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:143)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:98)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 2 methods
Detailed message:
Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@3d9aa364
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field com.sun.jna.internal.Cleaner.referenceQueue of constant
    com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3
  scanning root com.sun.jna.internal.Cleaner@41897cf3: com.sun.jna.internal.Cleaner@41897cf3 embedded in
    com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44)
  parsing method com.sun.jna.internal.Cleaner.getCleaner(Cleaner.java:44) reachable via the parsing context
    at static root method.(Unknown Source)

Error: Detected a started Thread in the image heap. Thread name: JNA Cleaner. Threads running in the image generator are no longer running at image runtime. If these objects should not be stored in the image heap, you can use

    '--trace-object-instantiation=com.sun.jna.internal.Cleaner$1'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.
The object was probably created by a class initializer and is reachable from a static field. You can request class initialization at image runtime by using the option --initialize-at-run-time=<class-name>. Or you can write your own initialization methods and call them explicitly from your main entry point.
Trace: Object was reached by
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$Node.waiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode@6f829e93
  reading field java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.firstWaiter of constant
    java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9: java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@460807c9
  reading field java.lang.ref.ReferenceQueue.notEmpty of constant
    java.lang.ref.ReferenceQueue@39cea108: java.lang.ref.ReferenceQueue@39cea108
  reading field java.lang.ref.Reference.queue of constant
    com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8: com.sun.jna.internal.Cleaner$CleanerRef@7d0c11f8
  reading field com.sun.jna.NativeLibrary.cleanable of constant
    com.sun.jna.NativeLibrary@193ee02e: Native Library <libc.dylib@13159884584>
  reading field com.sun.jna.Library$Handler.nativeLibrary of constant
    com.sun.jna.Library$Handler@144a3407: com.sun.jna.Library$Handler@144a3407
  reading field java.lang.reflect.Proxy.h of constant
    jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584>
  scanning root jdk.proxy4.$Proxy119@46a00bdd: Proxy interface to Native Library <libc.dylib@13159884584> embedded in
    org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:51)
  parsing method org.jline.terminal.impl.jna.osx.OsXNativePty.open(OsXNativePty.java:48) reachable via the parsing context
    at org.jline.terminal.impl.jna.JnaNativePty.open(JnaNativePty.java:63)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.open(JnaTerminalProvider.java:44)
    at org.jline.terminal.impl.jna.JnaTerminalProvider.newTerminal(JnaTerminalProvider.java:199)
    at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:814)
    at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:672)
    at ai.wanaku.cli.main.support.WanakuPrinter.terminalInstance(WanakuPrinter.java:136)
    at ai.wanaku.cli.main.support.ResponseHelper.commonResponseErrorHandler(ResponseHelper.java:12)
    at ai.wanaku.cli.main.commands.resources.ResourcesRemove.doCall(ResourcesRemove.java:38)
    at root method.(Unknown Source)


	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:129)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:809)
	... 6 more
------------------------------------------------------------------------------------------------------------------------
                        1.8s (9.0% of total time) in 105 GCs | Peak RSS: 2.50GB | CPU load: 7.31
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/build-artifacts.json (build_info)
 /Users/opiske/code/java/wanaku/cli/target/cli-0.0.6-SNAPSHOT-native-image-source-jar/cli-0.0.6-SNAPSHOT-runner-build-output-stats.json
```

--- a/cli/pom.xml
+++ b/cli/pom.xml
@@ -215,6 +215,20 @@
                 <skipITs>false</skipITs>
                 <quarkus.native.enabled>true</quarkus.native.enabled>
                 <quarkus.native.additional-build-args>
+                    <!-- initialize at build time -->
+                    --initialize-at-build-time=com.sun.jna.internal.Cleaner,
+                    --initialize-at-build-time=com.sun.jna.internal.Cleaner$CleanerThread,
+                    <!-- trace object instantiation -->
+                    --trace-object-instantiation=com.sun.jna.internal.Cleaner$CleanerThread,
+                    <!-- initialize at run time -->
+                    --initialize-at-run-time=org.jline.nativ.Kernel32,
+                    --initialize-at-run-time=com.sun.jna.Native,
+                    --initialize-at-run-time=com.sun.jna.Structure$FFIType,
+                    --initialize-at-run-time=org.jline.terminal.impl.jna.osx.OsXNativePty,
+                    --initialize-at-run-time=org.jline.terminal.impl.jna.linux.LinuxNativePty,
+                    --initialize-at-run-time=org.jline.terminal.impl.jna.linux.LinuxNativePty$UtilLibrary,
+                    --initialize-at-run-time=org.jline.terminal.impl.jna.freebsd.FreeBsdNativePty,
+                    --initialize-at-run-time=org.jline.terminal.impl.jna.solaris.SolarisNativePty,
                     --initialize-at-run-time=org.jline.nativ.Kernel32,
                     --initialize-at-run-time=org.jline.nativ.Kernel32$CONSOLE_SCREEN_BUFFER_INFO,
                     --initialize-at-run-time=org.jline.nativ.Kernel32$CHAR_INFO,

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -62,7 +62,7 @@
         <assertj-version>3.27.3</assertj-version>
         <juniversalchardet-version>1.0.3</juniversalchardet-version>
         <jansi-version>2.4.2</jansi-version>
-        <jna.version>5.11.0</jna.version>
+        <jna.version>5.17.0</jna.version>
         <awssdk.version>2.31.68</awssdk.version>
         <commons-logging.version>1.3.5</commons-logging.version>
 

